src/files/BrsFile.ts

```diff
@@ -41,20 +43,31 @@ export class BrsFile {
         this.pkgPath = s`${this.pkgPath}`;
         this.dependencyGraphKey = this.pkgPath.toLowerCase();
 
-        this.extension = path.extname(pathAbsolute).toLowerCase();
+        this.extension = extname(this.pkgPath).toLowerCase();
 
         //all BrighterScript files need to be transpiled
-        if (this.extension === '.bs') {
+        if (this.extension.endsWith('.bs')) {
             this.needsTranspiled = true;
         }
+        this.isTypedef = this.extension === '.d.bs';
+        if (!this.isTypedef) {
+            this.typedefKey = util.getTypedefPath(this.pathAbsolute);
```

Comment:
Feels a bit wasteful to calculate for every file - should it be undefined until it "has" a typedef?

Reply:
We need the key for every file right away because we use that key to subscribe to typedef file changes (add/update/delete of typedefs).